File: javascript/filters/single-line.js
/**
 * Output abbreviation on a single line (i.e. no line breaks)
 * @author Sergey Chikuyonok (serge.che@gmail.com)
 * @link http://chikuyonok.ru
 */
(function(){
	function process(tree, profile, level) {
		for (var i = 0, il = tree.children.length; i < il; i++) {
			/** @type {ZenNode} */
			var item = tree.children[i];
			if (item.type == 'tag') {
				// remove padding from item 
				var re_pad = /^\s+/;
				item.start = item.start.replace(re_pad, '');
				item.end = item.end.replace(re_pad, '');
			}
			
			// remove newlines 
			var re_nl = /[\n\r]/g;
			item.start = item.start.replace(re_nl, '');
			item.end = item.end.replace(re_nl, '');
			item.content = item.content.replace(re_nl, '');
			
			process(item);
		}
		
		return tree;
	}
	
	zen_coding.registerFilter('s', process);
})();

File: python/zencoding/filters/single-line.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
'''
Output abbreviation on a single line (i.e. no line breaks)
@author Sergey Chikuyonok (serge.che@gmail.com)
@link http://chikuyonok.ru
'''
import re
import zencoding

re_nl = re.compile(r'[\n\r]')
re_pad = re.compile(r'^\s+')

@zencoding.filter('s')
def process(tree, profile):
	for item in tree.children:
		if item.type == 'tag':
			# remove padding from item 
			item.start = re_pad.sub('', item.start)
			item.end = re_pad.sub('', item.end)
		
		# remove newlines 
		item.start = re_nl.sub('', item.start)
		item.end = re_nl.sub('', item.end)
		item.content = re_nl.sub('', item.content)
		
		process(item, profile)
	
	return tree